internal/testutils/asserts.go

```diff
@@ -0,0 +1,96 @@
+package testutils
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/getsentry/sentry-go/internal/otel/baggage"
+	"github.com/google/go-cmp/cmp"
+)
+
+func AssertEqual(t *testing.T, got, want interface{}, userMessage ...interface{}) {
+	t.Helper()
+
+	// Ideally, we would switch to cmp.Diff. However, in a general case, cmp.Diff
+	// is not able to compare structs with unexported (private) fields by default. There
+	// are ways to override modify that behaviour (e.g. by passing AllowUnexported), but
+	// it significantly complicates its usage.
+	if !reflect.DeepEqual(got, want) {
```

Comment:
Let's have a quick offline discussion about this, as I would like us to stick to either or 😄

Reply:
Yeah, it's a rabbit hole :/
Maybe you'll have some ideas, but at the moment it doesn't look very feasible to use `go-cmp` only.